core/editor.js

```diff
@@ -2228,3 +2305,53 @@ CKEDITOR.ELEMENT_MODE_INLINE = 3;
  * @event contentDomInvalidated
  * @param {CKEDITOR.editor} editor This editor instance.
  */
+
+/**
+ * Allow delayed editor creation (either with callback or intervals) if target element is detached from DOM.
+ *
+ *		config.delayIfDetached = true;
+ *
+ * @since 4.17.0
+ * @cfg {Boolean} [delayIfDetached=false]
+ * @member CKEDITOR.config
+ */
+CKEDITOR.config.delayIfDetached = false;
+
+/**
+ * Function with single argument. As argument is passed another function that continues editor creation.
+ * Allows to store the editor creation function and invoke it whenever it is convenient instead of periodically
+ * check if element is reattached to DOM.
+ *
+ * If this option is defined, editor will not run the default
+ * {@link CKEDITOR.config#delayIfDetached_interval interval checks}.
+ *
+ *		// Store the reference to the editor creation function.
+ *		var resumeEditorCreation;
+ *
+ *		config.delayIfDetached_callback = function( createEditor ) {
+ *			resumeEditorCreation = createEditor;
+ *		};
+ *
+ *		// Create editor calling `resumeEditorCreation()` whenever you choose (e.g. on button click).
+ *		resumeEditorCreation();
+ *
+ * @since 4.17.0
+ * @cfg {Function} [delayIfDetached_callback = function( createEditor ){ resumeEditorCreation = createEditor; }]
+ * @member CKEDITOR.config
+ */
+CKEDITOR.config.delayIfDetached_callback = function( createEditorFunction ) {
+	createEditorFunction();
+};
```

Comment:
Well - interval is used by default. It is determined by this line. https://github.com/ckeditor/ckeditor4/blob/16d640b8f19b97efd9b8478453493b4f1b2f0d04/core/editor.js#L1690

We are checking `config.delayIfDetached_callback` directly passed from the User - not `CKEDITOR.config.delayIfDetached_callback`. 

Provided default value actually based on this code: https://github.com/ckeditor/ckeditor4/blob/16d640b8f19b97efd9b8478453493b4f1b2f0d04/core/editor.js#L1695-L1700

This is because the entire 'merge' of configs happens later in... the editor constructor, which we don't want to call if creation needs to be delayed. So the default config option is useless after delay check. OTOH we need to present default behaviour of this option to the User.

Maybe we should extract this 'merge' at the beginning of the entire process?  :thinking:

Reply:
I would say this is a logic flaw here. If you take a look at [`CKEDITOR.config`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html) docs:

> Stores default configuration settings. Changes to this object are reflected in all editor instances, if not specified otherwise for a particular instance.

This means those lines declare default value which should be used if not override by the user. So the fact we use `config` does not work the same as the rest of the editor.

So:

* This declaration is default value. Which should be `null`/`undefined` here.
* We should check if `config.delayIfDetached_callback` was declared and use it instead of a default value - not sure if it will be better to move entire config merging logic here (probably it may break something :speak_no_evil:) or just check this specific value.

Reply:
> not sure if it will be better to move the entire config merging logic here (probably it may break something speak_no_evil)

Agree. It's very core-related stuff - so there may be various edge cases...

> We should check if config.delayIfDetached_callback was declared and use it instead of a default value

Just to be clear. We should change the default value to `undefined` & correct API docs comments to describe that we use this callback only if it's declared. Otherwise, we use our default logic - which is interval checks?

Reply:
> Just to be clear. We should change the default value to `undefined` & correct API docs comments to describe that we use this callback only if it's declared. Otherwise, we use our default logic - which is interval checks?

I mean, if the callback is not declared it won't be use because there is nothing to use there - so it will work the same as other configuration options which don't have value by default. So yes, the value here should be `undefined` as we want interval to be used by default. And I think the docs are already fine now :+1:

Reply:
And as for merging, since we change logic here, it will be good to cover with unit test too.